File: Question2 Programs/MMI.cpp
#include <iostream>
#include <vector>
#include <ctime>

using namespace std;

// Function to multiply two matrices of integers
void MatrixMultiplicationInt(int N) {
    vector<vector<int>> A(N, vector<int>(N, 1));
    vector<vector<int>> B(N, vector<int>(N, 1));
    vector<vector<int>> C(N, vector<int>(N, 0));
    
    struct timespec start, end;
    clock_gettime(CLOCK_MONOTONIC, &start);

    for(int i = 0; i < N; ++i) {
        for(int j = 0; j < N; ++j) {
            for(int k = 0; k < N; ++k) {
                C[i][j] += A[i][k] * B[k][j];
            }
        }
    }

    clock_gettime(CLOCK_MONOTONIC, &end);
    double time_taken = (end.tv_sec - start.tv_sec) + (end.tv_nsec - start.tv_nsec) / 1e9;
    cout << "Time taken by integer multiplication: " << time_taken << " seconds" << endl;
}

int main() {
    int N = 64;

    cout << "Matrix Size: " << N << endl;
    MatrixMultiplicationInt(N);
    cout << "-----------------------" << endl;

    return 0;
}

File: Question2 Programs/MMI.py
import time

# Function to multiply two matrices of integers
def MatrixMultiplicationInt(N):
    A = [[1 for _ in range(N)] for _ in range(N)]
    B = [[1 for _ in range(N)] for _ in range(N)]
    C = [[0 for _ in range(N)] for _ in range(N)]

    start_time = time.time()

    for i in range(N):
        for j in range(N):
            for k in range(N):
                C[i][j] += A[i][k] * B[k][j]

    end_time = time.time()
    print(f"Time taken by integer multiplication: {end_time - start_time:.5f} seconds")

if __name__ == "__main__":
    N = 64
    print(f"Matrix Size: {N}")
    MatrixMultiplicationInt(N)
    print("-----------------------")
